File: mod/src/client/java/top/fifthlight/armorstand/extension/RenderPassExt.java
package top.fifthlight.armorstand.extension;

import org.jetbrains.annotations.NotNull;
import top.fifthlight.armorstand.render.VertexBuffer;

public interface RenderPassExt {
    void armorStand$setVertexBuffer(@NotNull VertexBuffer vertexBuffer);
    void armorStand$draw(int baseVertex, int firstIndex, int count, int instanceCount);
}

File: mod/src/client/kotlin/top/fifthlight/armorstand/extension/RenderPassExt.kt
package top.fifthlight.armorstand.extension

import com.mojang.blaze3d.systems.RenderPass
import top.fifthlight.armorstand.render.VertexBuffer

fun RenderPass.setVertexBuffer(vertexBuffer: VertexBuffer) =
    (this as RenderPassExt).`armorStand$setVertexBuffer`(vertexBuffer)

fun RenderPass.draw(baseVertex: Int, firstIndex: Int, count: Int, instanceCount: Int) =
    (this as RenderPassExt).`armorStand$draw`(baseVertex, firstIndex, count, instanceCount)